odtp/cli/execution.py

```diff
@@ -105,6 +109,41 @@ def output(
         print(f"ERROR: Output printing failed: {e}")       
         raise typer.Abort()
 
+@app.command()
+def delete(
+    execution_name: str = typer.Option(
+        None, "--execution-name", help="Specify the name of the execution"
+    ),
+    execution_id: str = typer.Option(
+        None, "--execution-id", help="Specify the ID of the execution"
+    ),
+    project_path: str = typer.Option(
+        ..., "--project-path", help="Specify the path for the execution"
+    ),
+):
+    try:
+        if execution_id is None and execution_name is None:
+            raise typer.Exit("Please provide either --execution-name or --execution-id")
+
+        if execution_name:
+            execution_id = db.get_document_id_by_field_value("title", execution_name, "executions")
+
+        # S3
+        s3_keys = db.get_all_outputs_s3_keys(execution_id)
+        s3M = s3Manager()
+        s3M.deletePaths(s3_keys)
+
+        # DB
+        db.delete_execution(execution_id)
+
+        # Folders
+        odtp_env.delete_folder(project_path)
```

Comment:
Maybe we should making the deletion of the folder optional. As long as the project-path is not part of the execution I feel it is better to not have it on the deletion of executions either.

Reply:
Fair!